sample-apps/dotnet/asp_frontend_service/Controllers/AppController.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Amazon.S3;
+using Microsoft.AspNetCore.Mvc;
+
+namespace asp_frontend_service.Controllers;
+
+[ApiController]
+[Route("[controller]")]
+public class AppController : ControllerBase
+{
+    // reads and writes to bool are atomic according to spec:
+    // https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/variables#96-atomicity-of-variable-references
+    private static bool shouldSendLocalRootClientCall = false;
+    private static bool threadStarted = false;
+    private readonly AmazonS3Client s3Client = new AmazonS3Client();
+    private readonly HttpClient httpClient = new HttpClient();
+
+    private static readonly Thread thread = new Thread(() =>
+            {
+                while (true)
+                {
+                    if (shouldSendLocalRootClientCall)
+                    {
+                        try
+                        {
+                            shouldSendLocalRootClientCall = false;
+                            // forcing the new activity to not have a parent and thus become a local root span
+                            Activity.Current = null;
+                            var localHttpClient = new HttpClient();
+                            _ = localHttpClient.GetAsync("http://local-root-client-call").Result;
+                        }
+                        catch (Exception)
+                        {
+                        }
+                    }
+
+                    Thread.Sleep(500);
+                }
+            });
+
+    public AppController()
+    {
+        if (!threadStarted)
+        {
+            Console.WriteLine("Starting thread");
+            threadStarted = true;
+            thread.Start();
+        }
+    }
+
+    [HttpGet]
+    [Route("/outgoing-http-call")]
+    public string OutgoingHttp()
+    {
+        _ = this.httpClient.GetAsync("https://aws.amazon.com").Result;
+
+        return this.GetTraceId();
+    }
+
+    [HttpGet]
+    [Route("/aws-sdk-call")]
+    public string AWSSDKCall()
+    {
+        _ = this.s3Client.ListBucketsAsync().Result;
```

Comment:
If this is async, shouldn't it generate similar telemetry to `/client-call` API? Is this actually an async operation?

Reply:
It works actually